sdk/nodejs/provider/experimental/analyzer.ts

```diff
@@ -584,6 +588,26 @@ function isInput(type: typescript.Type): boolean {
     return hasOutput && hasPromise && hasOther;
 }
 
+function isResourceReference(type: typescript.Type, checker: typescript.TypeChecker): boolean {
+    if (type.isClassOrInterface()) {
+        const baseTypes = checker.getBaseTypes(type as typescript.InterfaceType);
+        for (const baseType of baseTypes) {
+            const symbol = baseType.getSymbol();
```

Comment:
Will we catch a case when MyResource inherits MyAbstractResource inherits CustomResource? Not sure it's very important but curious.